src/ring_theory/fintype.lean

```diff
@@ -0,0 +1,28 @@
+/-
+Copyright (c) 2020 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Morrison
+-/
+import algebra.ring
+import data.fintype.basic
+
+/-!
+Some facts about finite rings.
+
+(There may be a better home for these elsewhere?)
+-/
+
+lemma zero_not_mem_range_coe_units (R : Type*) [nonzero_semiring R] :
```

Comment:
This lemma doesn't need `fintype`. BTW, do we have `set.range (coe : units R → R) = {x | is_unit x}` (AFAIR not `rfl` because of different LHS/RHS in definitions) and `zero_not_is_unit`? And should we change `is_unit` to match `set.range`?

Reply:
We have the latter but it assumes `nonzero_comm_ring` instead of `nonzero_semiring`. Could you fix it, please?

Reply:
I've generalised `not_is_unit_zero` appropriately.

Reply:
I've also flipped the equality in the definition of `is_unit`. So far it removes more `.symm`s downstream than it adds (but is still building...).